src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPDocumentMappingProvider.cs

```diff
@@ -58,5 +70,148 @@ public async override Task<RazorMapToDocumentRangeResponse> MapToDocumentRangeAs
 
             return documentMappingResponse;
         }
+
+        public async override Task<WorkspaceEdit> RemapWorkspaceEditAsync(WorkspaceEdit workspaceEdit, CancellationToken cancellationToken)
+        {
+            if (workspaceEdit == null)
+            {
+                return workspaceEdit;
+            }
+            else if (workspaceEdit.DocumentChanges != null)
+            {
+                // The LSP spec says, we should prefer `DocumentChanges` property over `Changes` if available.
+                var remappedEdits = await RemapVersionedDocumentEditsAsync(workspaceEdit.DocumentChanges, cancellationToken).ConfigureAwait(false);
+                return new WorkspaceEdit()
+                {
+                    DocumentChanges = remappedEdits
+                };
+            }
+            else
```

Comment:
`workspaceEdit.Changes` null check required?

Reply:
Good point. It is slightly unrealistic for Roslyn to send both of these as null but we can't make assumptions 👍

Reply:
Going to do this in my next PR.